src/v/kafka/server/handlers/fetch.cc

```diff
@@ -1350,6 +1351,28 @@ class simple_fetch_planner final : public fetch_planner::impl {
                   return;
               }
 
+              /**
+               * in sanction mode (without an enterprise license), the audit log
+               * topic is not consumable
+               */
+              if (unlikely(
+                    octx.rctx.feature_table().local().should_sanction()
+                    && fp.topic_partition.get_topic()
+                         == model::kafka_audit_logging_topic)) {
+                  thread_local static ss::logger::rate_limit rate(1s);
+                  vloglr(
+                    klog,
+                    ss::log_level::warn,
+                    rate,
+                    "An enterprise license is required to consume the "
+                    "audit log topic");
```

Comment:
Just an FYI. I was digging a bit on the seastar code to see what the rate_limit is doing and how it is applied.

TL;DR: This is the correct usage for limited rate of messages :+1: 

When a `.log(...)` call has a `rate_limit` object attached to it, the `rate_limit` object is queried to see if enough time has passed.
I.e. if we have two different `.log(...)` calls, on the same logger, with two independent `rate_limit` objects attached to each, they will both be printing independently of each other at the correct intervals. If we were to pass the same `rate_limit` object to 2 or more different `.log(...)` calls, then we would get a single logging event (whichever happens to be) out of all, every `rate_limit` time.